com.unity.render-pipelines.high-definition/Editor/Wizard/HDProjectSettings.cs

```diff
@@ -34,9 +44,51 @@ enum Version
         bool m_WizardNeedRestartAfterChangingToDX12 = false;
         [SerializeField]
         bool m_WizardNeedToRunFixAllAgainAfterDomainReload = false;
+
+        // Note: this is for the *material version*, which uses MaterialPostProcessor k_Migrations[] as migration
+        // functions table. See note on Version enum above.
         [SerializeField]
         int m_LastMaterialVersion = k_NeverProcessedMaterialVersion;
 
+        [SerializeField]
+        ShaderGraphVersion m_HDShaderGraphLastSeenVersion = ShaderGraphVersion.NeverMigrated;
+
+        // This tracks latest versions of every plugin subtargets ever seen in the project.
+        // The dictionary key is the SubTarget GUID.
+        // If the project has ever seen an external-to-HDRP ("plugin") material, it saves the version
+        // of the material here. That way, we can use the same logic to forward upgrade calls
+        // than for the internally known HDRP materials: if a plugin material version is higher than
+        // what is seen here, we can trigger a re-import (see RegisterUpgraderReimport in MaterialPostProcessor).
+        [SerializeField]
+        PluginMaterialVersions m_PluginMaterialVersions = new PluginMaterialVersions();
+
+        // Same as above but for the *subtarget* versioning specific to the subtarget (ie not HDRP ShaderGraphVersion,
+        // see m_HDShaderGraphLastSeenVersion above for that).
+        [SerializeField]
+        PluginMaterialVersions m_PluginSubTargetVersions = new PluginMaterialVersions();
+
```

Comment:
The m_PluginSubTargetVersions dictionary is for the same purpose as m_HDShaderGraphLastSeenVersion but for plugins. See also s_AutoReimportProjectShaderGraphsOnVersionUpdate in the MaterialPostProcessor.